ui/src/app/pages/admin/workspace/about-publish-consent-modal.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import { Button } from 'app/components/buttons';
+import { SemiBoldHeader } from 'app/components/headers';
+import { CheckBox } from 'app/components/inputs';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import colors from 'app/styles/colors';
+
+interface Props {
+  onConfirm: () => void;
+  onCancel: () => void;
+}
+export const AboutPublishConsentModal = (props: Props) => {
+  const [checkboxStates, setCheckboxStates] = useState({
+    checkbox1: false,
+    checkbox2: false,
+    checkbox3: false,
```

Comment:
Could you please update these to have more descriptive?

I think it might be simpler, if they were all separate state variables:

` const [noHarmfulLanguageAcknowledged, setNoHarmfulLanguageAcknowledged] = useState(false)`
 
 Then you could get rid of handleCheckboxChange in favor of something like this:
 
 `          <CheckBox
            style={{ marginRight: '0.375rem' }}
            onChange={(checked) => setNoHarmfulLanguageAcknowledged(checked)}
            checked={noHarmfulLanguageAcknowledged}
          />`

Reply:
I think having them in one seems better in that we can do sth like this
```
  const showConfirmButton = Object.values(checkboxStates).every(
    (state) => state === true
  );
```
And if later we add another checkbox, we don't need to change the logic